cmd/retester/main.go

```diff
@@ -68,6 +71,7 @@ func gatherOptions() options {
 
 	fs.BoolVar(&o.dryRun, "dry-run", true, "Dry run for testing. Uses API tokens but does not mutate.")
 	fs.BoolVar(&o.runOnce, "run-once", false, "If true, run only once then quit.")
+	fs.BoolVar(&o.cacheFileOnS3, "cache-file-on-s3", false, "If true, use aws s3 bucket to store the cache file.")
```

Comment:
Could we invalidate if `cacheFileOnS3` and `cache-file` are both (or neither) set?
Also add the unit tests for those new cases?

Reply:
After answering you in slack, I gave another thought.

>  we always need it (whether cacheFileOnS3 is true or false)

`cache-file` is not set if we want to disable the cache. It is unlikely we do it in our production but i think it is a valid case.

Before this PR, `cache-file` has two functions:

1. cache is enabled _iff_ it is not empty string.
2. it provides the name of the file.

After this PR, `cache-file`'s functions more or less the same:
1. cache is enabled _iff_ it is not empty string. (same as before)
2. it provides the name of the file: either local or on s3 according to `cache-file-on-s3`

So the only invalid case that I can think of is that `cache-file` is not set but `cache-file-on-s3` is `true`.
Let us error out in this case.